src/main/java/com/jcraft/jsch/Logger.java

```diff
@@ -40,6 +43,20 @@ LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
   public boolean isEnabled(int level);
 
   public void log(int level, String message);
+  
+  public default void log(int level, String message, Throwable cause) {
+    if (!isEnabled(level)) {
+      return;
+    }
+    if (cause != null) {
+      StringWriter sw = new StringWriter();
+      try (PrintWriter pw = new PrintWriter(sw, true)) {
+        cause.printStackTrace(pw);
+      }
+      message += "\r\n" + sw.toString();
```

Comment:
I think we should do this instead:
```
message += System.lineSeparator() + sw.toString();
```

That way it uses the correct line separator for whatever platform is in use.

Reply:
I also wonder if we should override this default method in the concrete implementations I added to utilize their specialized methods for logging exceptions?

Reply:
I've changed the line break and added specific log-methods to framework-specific implementations. I've also added test cases for all the logging frameworks that allow logger configuration in a programmatic way, i.e. there is no test for Log4j2.